misc/dockerfiles/run.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+set -e
+
+echo_green() { echo -e "\e[42m$@\e[0m"; }
+echo_red()   { echo -e "\e[101m$@\e[0m"; }
+
+echo_exec() { echo_green "$@"; "$@"; }
+
+DIR="$(cd "$(dirname $0)"; pwd)"
+ROOT_DIR="$(dirname $(dirname $DIR))"
```

Comment:
I don't really understand what you're doing here - what's the `cd` and `pwd` for? Wouldn't something like `root_dir="$(readlink -f "${0%/*}/../..")"` or so work? (see [parameter expansion](http://wiki.bash-hackers.org/syntax/pe#substring_removal))

Reply:
I've tested different methods at work to handle getting the full path from the `$0` with edge cases like when it's in the `$PATH` or called directly with `./` etc. and this is in my opinion the best mixture between readability and functionality. Since `$()` is called in a sub-process it does not change the current directory of the script.